From 42a0e83df41494cc67157f98b18a4087f6853911 Mon Sep 17 00:00:00 2001
From: Andre Fernando Zanella Kantek
 <AndreFernandoZanella.Kantek@windriver.com>
Date: Thu, 1 Sep 2022 10:05:09 -0300
Subject: [PATCH] Use prefixlen for IPv6 address operations

use the prefix length variable available to the script to correctly
set the address passed by the dhclient program.


Signed-off-by: Andre Fernando Zanella Kantek <AndreFernandoZanella.Kantek@windriver.com>
---
 debian/dhclient-script.linux | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/debian/dhclient-script.linux b/debian/dhclient-script.linux
index 8b29988..915cc13 100644
--- a/debian/dhclient-script.linux
+++ b/debian/dhclient-script.linux
@@ -403,7 +403,7 @@ case "$reason" in
 
         if [ "${new_ip6_address}" ]; then
             # set leased IP
-            ip -6 addr add ${new_ip6_address} \
+            ip -6 addr add ${new_ip6_address}/${new_ip6_prefixlen} \
                 dev ${interface} scope global
         fi
 
@@ -429,7 +429,7 @@ case "$reason" in
         fi
 
         # delete leased IP
-        ip -6 addr del ${old_ip6_address} \
+        ip -6 addr del ${old_ip6_address}/${old_ip6_prefixlen} \
             dev ${interface}
 
         ;;
-- 
2.17.1

